doc/introduction/measurements.rst

```diff
@@ -524,6 +524,61 @@ Collecting statistics for sequences of mid-circuit measurements is supported wit
     When collecting statistics for a list of mid-circuit measurements, values manipulated using
     arithmetic operators should not be used as this behaviour is not supported.
 
+Configuring mid-circuit measurements
+************************************
+
+As seen above, there are multiple ways in which circuits with mid-circuit measurements can be executed with
+PennyLane. For ease of use, we provide the following configuration options to users when initializing a
+:class:`~pennylane.QNode`:
+
+* ``mcm_method``: To set the method used for applying mid-circuit measurements. Use ``mcm_method="deferred"``
+  to use the deferred measurements principle or ``mcm_method="one-shot"`` to use the one-shot transform as
+  described above.
+
+  .. warning::
+
+      If the ``mcm_method`` argument is provided, the :func:`~pennylane.defer_measurements` or
+      :func:`~pennylane.dynamic_one_shot` transforms should not be applied directly to a :class:`~pennylane.QNode`
+      as it can lead to incorrect behaviour.
+
+* ``postselect_mode``: To configure how invalid shots are handled when postselecting mid-circuit measurements
+  with finite-shot circuits. Use ``postselect_mode="hw-like"`` to discard invalid shots. In this case, the number
+  of samples that are used for processing results will be less than or equal to the total number of shots. Use
+  ``postselect_mode="fill-shots"`` to keep invalid shots.
+
+  .. note::
+
+      If ``postselect_mode="fill-shots"``, the specified ``mcm_method`` will impact the results due to the particular
+      features/limitations of the requested ``mcm_method``.
+
+      * If using ``mcm_method="hw-like"``, invalid samples will be replaced with ``np.iinfo(np.int32).min``,
+        and these invalid values will not be used for processing final results.
+      * If using ``mcm_method="deferred"``, all shots will be projected to the postselected value, so all
+        shots will be considered valid.
```

Comment:
This is not what I expect to be happening :thinking: `postselect_mode="fill-shots"` would mean that you always get valid shots, but you keep going until you get the number of shots requested by the user. We can talk offline (e.g., after the standup) on how to best word this.